examples/platforms/utils/link_metrics.cpp

```diff
@@ -257,4 +257,11 @@ uint8_t otLinkMetricsEnhAckGetDataLen(const otMacAddress *aMacAddress)
 exit:
     return len;
 }
+
+void otLinkMetricsResetEnhAckProbing()
+{
+    // clean the pool and delete all the content of the list
+    GetLinkMetricsDataInfoPool().FreeAll();
+    GetLinkMetricsDataInfoActiveList().Clear();
```

Comment:
```suggestion
    GetLinkMetricsDataInfoActiveList().Clear();
    GetLinkMetricsDataInfoPool().FreeAll();
```
I think the order doesn't matter here. But generally it would be better to first remove resource's reference and then free the resource.